### DISCUSSION TITLE
Firefly III behind Nginx Proxy Manager (NPM)

### DISCUSSION THREAD

**Discussion by Jandokus** [firefly-iii/firefly-iii#8614] [Firefly III]
I've been running Firefly III locally for a month now and it works fine. But I would like to be able to access Firefly III from the internet. I thought I followed the documentation and all the tips. However, I can't get it done.

Now when I want to access Firefly III (behind Nginx Proxy Manager), I get a page with 'Welcome to OpenResty!'.
Perhaps you can help me to indicate what I am doing wrong.

For completeness, I have added some images of my configuration below.
![IMG_7931](https://github.com/firefly-iii/firefly-iii/assets/1504140/f47f575f-7e1f-4f3b-b2e4-ec76e94b3357)
![IMG_7930](https://github.com/firefly-iii/firefly-iii/assets/1504140/bb39362a-ec4d-40c6-bd5c-05f54241acc8)
![IMG_7929](https://github.com/firefly-iii/firefly-iii/assets/1504140/18a87750-59bb-4138-b5a0-3a8662cb48d5)
![IMG_7934](https://github.com/firefly-iii/firefly-iii/assets/1504140/b0adfdf4-d24a-446b-a86e-5743dd539168)
![IMG_7933](https://github.com/firefly-iii/firefly-iii/assets/1504140/06eef71f-25be-4d81-8e42-df008fe5fd8f)

**Comment by JC5** [comment]
I know the Openresty page is probably part of the Nginx Proxy Manager (Firefly III doesn't use it). Could it be the host name is incorrect? A basic thing, but it's all I have right now.

**Reply by Jandokus** [reply]
@JC5, I have about 15 other web applications running under the same hostname, all of which work. That's why I find it so strange. Thank you for your thoughts. If you come up with anything else, I'd love to hear it.

**Reply by JC5** [reply]
Same, but since you get sent to a default openresty page I suspect there's a difference between Firefly III's config and those other apps.

**Comment by cachho** [comment]
I use it with NginxProxyManager with no problems. Used it with wildcard certificates and regular certificates too. I'm not using custom locations or advanced config, it works out of the box.

**Comment by Jandokus** [comment]
@cachho , thanks for your response. This puts me one step further. I also deleted the custom locations or advanced config. If I now enter the URL https://fireflyiii.hostname.nl:7070, I end up at OpenResty (port number 7070 is fictitious, but I don't use 443 for https!).

When I refresh the page, I see that the URL https://fireflyiii.hostname.nl/login is being accessed (without port number) and a message states that the server is not responding.
If I manually insert the port number (so https://fireflyiii.hostname.nl:7070/login) I get the desired Firefly III login page. However, after that I can't get any further. Probably because Firefly III cannot handle a https port number other than 443. Or has that been concluded too quickly? Is there a solution for this?

**Reply by JC5** [reply]
Make sure you also forward the port to Firefly III, otherwise it will indeed default to 443 or 80 (or rather, Apache will).

**Reply by lemoncat7** [reply]
I also encountered this problem, but I'm not sure how to do it

**Comment by ilikecola** [comment]
I am using NPM and got the same error, but working fine with IP and no proxy.

If there is no custom nginx configuration, the URL https://hostname/login is being accessed (without port number).
I tried some configuration,like:
```
location / {
    proxy_set_header X-Forwarded-Host $host:$port;
    proxy_set_header X-Forwarded-Server $host:$port;
}
```
But getting a page with 'Welcome to OpenResty!'.
Any other proxied server is working fine.

I dug some headers in traffic and found interesting things that the location in the header redirected to 'login' was wrong, which is without the port number and static files' URL is also without the port number.

**Comment by Sinterdial** [comment]
My NPM advance configuration: 
```
location / {
    proxy_set_header X-Forwarded-Host $host;
    proxy_set_header X-Forwarded-Server $host;
    proxy_set_header X-Forwarded-Proto $scheme;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header Host $host;
    client_max_body_size 64M;
    proxy_read_timeout 300s;
}
``` 
But also getting a page with Welcome to OpenResty

![image](https://github.com/user-attachments/assets/ae2fdaf4-d576-4894-a820-b96d052f436f)

the only item working fine seem to be favicon

**Comment by GuideGlyph** [comment]
If someone have this problem as me today and spend to much time struggle. check https://github.com/firefly-iii/firefly-iii/issues/2319

this may help:
firefly docker env `TRUSTED_PROXIES` set to `**`

**Comment by Unsantae** [comment]
By any chance are you running firefly and NPM from a docker container?  If so, your NPM container and docker container need to be part of the same docker network.  I’m running behind NPM and sorting out that was the only problem I had.

**Comment by Sinterdial** [comment]
it works fine for me when i added a location setting in npm
![1000087771](https://github.com/user-attachments/assets/e5db37a4-d5bc-496c-b3db-7038255b4b8c)
